src/Browser.php

```diff
@@ -169,49 +171,111 @@ public function getVersion()
     }
 
     /**
-     * Set the Browser to be a robot.
+     * Detects scripted agents (robots / bots)
+     * Returns a resolved ScriptedAgent object if detected.
+     * Otherwise returns false.
      *
-     * @param bool $isRobot
+     * @return ScriptedAgent|bool
+     */
+    public function detectScriptedAgent()
```

Comment:
The reasoning here is that detectScriptedAgent can provide a replacement for $isRobot in this class.  This function provides a quick pre-scan to avoid in-depth substring searches in ScriptedAgent if not necessary.